lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4916,6 +4916,17 @@ bool LLVMToSPIRVBase::transExecutionMode() {
       case spv::ExecutionModeNumSIMDWorkitemsINTEL:
       case spv::ExecutionModeSchedulerTargetFmaxMhzINTEL:
       case spv::ExecutionModeMaxWorkDimINTEL:
+      case spv::ExecutionModeRegisterMapInterfaceINTEL: {
+        if (!BM->isAllowedToUseExtension(
+                ExtensionID::SPV_INTEL_kernel_attributes))
+          break;
+        AddSingleArgExecutionMode(static_cast<ExecutionMode>(EMode));
+        BM->addExtension(ExtensionID::SPV_INTEL_kernel_attributes);
+        // CapabilityFPGAKernelAttributesv2INTEL implicitly defines
+        // CapabilityFPGAKernelAttributesINTEL
+        BM->addCapability(CapabilityFPGAKernelAttributesINTEL);
+        BM->addCapability(CapabilityFPGAKernelAttributesv2INTEL);
```

Comment:
Not quite correct, `CapabilityFPGAKernelAttributesv2INTEL` is needed only for `RegisterMapInterfaceINTEL` and not for other execution modes.

Reply:
Are you suggesting I change the comment here or the code?

Reply:
The code, see below